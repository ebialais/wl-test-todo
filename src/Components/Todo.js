import React from 'react';
import './style.css';

const Todo = (props) => {
  const todo = props.todo
  
  return (
    <li className={ todo.completed ? 'completed todo-element' : 'uncompleted todo-element'}  >
      <p className={ todo.completed ? 'completed-todo-p' : 'todo-p'}>{todo.action}</p>
      <div className='todo-buttons-wrapper'>
        <button
          className='todo-button'
          onClick={()=>props.toggleCompleted(todo.id)}
        >
          <i className="fi fi-br-check"></i>
        </button>
        <button 
          className='todo-button'
          onClick={()=>props.deleteTodo(todo.id)}
        >
          <i className="fi fi-rs-trash"></i>
        </button>
      </div>
    </li>
  )
};

export default Todo;
