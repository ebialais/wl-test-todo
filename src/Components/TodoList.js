import React from 'react'
import Todo from './Todo'
import './style.css';


const TodoList = (props) => {
    
    const todos = props.todos;

    const toggleCompleted = (id) => {
        props.toggleCompleted(id)
    };

    const deleteTodo = (id) => {
        props.deleteTodo(id)
    }
    
    return (
        <ul className='todo-list'>
            {todos.map((todo) => {
                return (
                    <Todo
                        key={todo.id} 
                        todo={todo}
                        toggleCompleted={toggleCompleted}
                        deleteTodo={deleteTodo}
                    />
                )
            })}
        </ul>
    )
};

export default TodoList;