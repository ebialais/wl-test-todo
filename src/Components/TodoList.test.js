import React from 'react';
import { act  } from '@testing-library/react';
import ShallowRenderer from 'react-test-renderer/shallow';
import TodoList from './TodoList';
import Todo from './Todo';

test('test todoList rendering', async () => {
    await act(async () => {
        const mockTodos = [
            { id: 1, action: 'first stuff to do', completed: false },
            { id: 2, action: 'second stuff to do', completed: false },
            { id: 3, action: 'third stuff to do', completed: false },
        ];
        const renderer = new ShallowRenderer();
        const renderedTodoList = renderer.render( <TodoList todos={mockTodos}/> );
        expect(renderedTodoList).toMatchSnapshot('rendered TodoList');
    });
});

test('list should render 3 elements', async () => {
    await act(async () => {
        const mockTodos = [
            { id: 1, action: 'first stuff to do', completed: false },
            { id: 2, action: 'second stuff to do', completed: false },
            { id: 3, action: 'third stuff to do', completed: false },
        ];
        const renderer = new ShallowRenderer();
        const renderedTodoList = renderer.render(<TodoList todos={mockTodos}/>);
        expect(renderedTodoList.props.children).toHaveLength(3);
    });
});

