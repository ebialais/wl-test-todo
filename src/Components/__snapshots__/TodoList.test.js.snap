// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test todoList rendering: rendered TodoList 1`] = `
<ul
  className="todo-list"
>
  <Todo
    role="listitem"
    todo={
      Object {
        "action": "first stuff to do",
        "completed": false,
        "id": 1,
      }
    }
  />
  <Todo
    role="listitem"
    todo={
      Object {
        "action": "second stuff to do",
        "completed": false,
        "id": 2,
      }
    }
  />
  <Todo
    role="listitem"
    todo={
      Object {
        "action": "third stuff to do",
        "completed": false,
        "id": 3,
      }
    }
  />
</ul>
`;
